### PR TITLE
Test smarthost settings

### PR DESCRIPTION
**Pull request by stephdl**
- Validate if the smarthost configuration is good by sending an email
- Display the password input if you press a button
- add info box the smtps over 464 is not supported

Screenshot (75)
Screenshot (74)
Screenshot (73)

https://github.com/NethServer/dev/issues/6445

### CODE REVIEW

File: api/system-settings/execute [hunk 1]
```diff
@@ -24,21 +24,38 @@
...
         ;;
+    "test-smarthost")
+        hostname=$(echo $data | jq -r '.SmartHostName')
+        username=$(echo $data | jq -r '.SmartHostUsername')
+        password=$(echo $data | jq -r '.SmartHostPassword')
+        port=$(echo $data | jq -r '.SmartHostPort')
+        tls=$(echo $data | jq -r '.SmartHostTlsStatus')
+        [[ $tls = 'true' ]] && tls='--ssl' || tls=''
+        /usr/bin/curl smtp://$hostname:$port -v  --connect-timeout 10 --mail-from $username --mail-rcpt $username --user "$username:$password" $tls <<EOF
+Subject: Test smarthost credentials: just ignore
```
Comment (gsanchietti) [inline api/system-settings/execute:52]:
```suggestion
Subject: Test smarthost credentials: just ignore
```
```suggestion
Subject: Test smarthost credentials
```

File: api/system-settings/execute [hunk 2]
```diff
@@ -24,21 +24,38 @@
...
+Date: $(date -R)
+Message-ID: testEmail.$(date +%s)@$(hostname -d)
+From: no-reply@$(hostname -d)
+To: $username
+
+We are testing the smarthost credentials
```
Comment (gsanchietti) [inline api/system-settings/execute:58]:
```suggestion
We are testing the smarthost credentials
```
```suggestion
Test mail for smarthost credentials.
```

File: ui/public/i18n/language.json [hunk 3]
```diff
@@ -896,6 +896,8 @@
     "user_settings_page_urls_explain": "These URLs are obtained by server aliases and IP addresses on green and red interfaces.",
     "password_updated_error_1": "Password not changed. Ensure that",
     "password_updated_error_2": "old password",
+    "smarthost_test_configuration": "Configuration",
+    "smarthost_send_an_email": "Test mail",
```
Comment (gsanchietti) [inline ui/public/i18n/language.json:900]:
```suggestion
    "smarthost_send_an_email": "Test mail",
```
```suggestion
    "smarthost_send_an_email": "Send test mail",
```

File: ui/public/i18n/language.json [hunk 4]
```diff
@@ -1257,6 +1259,7 @@
     "disaster_recovery": "Disaster recovery",
     "user_settings_page": "Allow users to change their password and other settings on a web page outside Cockpit (on port 443). This feature can be enabled only if 'Override the shell of users' is enabled as well.",
     "OtpR2WOath_Auth": "The VPN policy requires the usage of OTP to access the network.",
+    "smarthost_encrypt": "The Encryption over smtps (default port 465) is not supported, only smartTls (default port 587) is supported",
```
Comment (gsanchietti) [inline ui/public/i18n/language.json:1262]:
```suggestion
    "smarthost_encrypt": Encryption on port 465 (SMTPS) is not supported, only StartTLS is supported (default port 587)",
```
```suggestion
    "smarthost_encrypt": "The Encryption over smtps (default port 465) is not supported, only smartTls (default port 587) is supported",
```

**Review by gsanchietti — COMMENTED** [review]
The implementation looks very good, I'd change only some labels.

**Comment by stephdl** [comment]
change done, test ok on my side

File: ui/public/i18n/language.json [hunk 5]
```diff
@@ -1257,6 +1259,7 @@
     "disaster_recovery": "Disaster recovery",
     "user_settings_page": "Allow users to change their password and other settings on a web page outside Cockpit (on port 443). This feature can be enabled only if 'Override the shell of users' is enabled as well.",
     "OtpR2WOath_Auth": "The VPN policy requires the usage of OTP to access the network.",
+    "smarthost_encrypt": Encryption on port 465 (SMTPS) is not supported, only StartTLS is supported (default port 587)",
```
Comment (gsanchietti) [inline ui/public/i18n/language.json:1262]: Wrong JSON format, missing `"`

**Review by gsanchietti — CHANGES_REQUESTED** [review]

File: ui/public/i18n/language.json (hunk 5 above)
Comment (stephdl) [reply ui/public/i18n/language.json]: I know I corrected it locally

**Review by gsanchietti — APPROVED** [review]
Looks good to me.

**Pull request merged**